presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementationHeader.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar.annotations;
+
+import com.facebook.presto.operator.Description;
+import com.facebook.presto.operator.scalar.ScalarFunction;
+import com.facebook.presto.operator.scalar.ScalarHeader;
+import com.facebook.presto.operator.scalar.ScalarOperator;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.FunctionRegistry.mangleOperatorName;
+import static com.google.common.base.CaseFormat.LOWER_CAMEL;
+import static com.google.common.base.CaseFormat.LOWER_UNDERSCORE;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class ScalarImplementationHeader
```

Comment:
`ScalarHeader` is used in `ParametricScalar` too.

To clairfy:
`ScalarImplementationHeader` is all data obtained from parsed code and annotations. It describe function that can have multiple specializations so it does not contains argument information as `Signature` does.
`ScalarHeader` is "metadata" of ScalarMethod that is obtained from annotations which is not part of `Signature` (in fact it's `ScalarImplementationHeader - name`).
`Signature` is what functions system requires but nothing more than that (includes `name`).

This is separated that way mostly to avoid duplicate of `name` inside `ParametricScalar` (both `ScalarImplementationHeader` and `Signature` do contain `name`). `Signature` is forced there by the rest of the functions system.
Naming is sketchy here especially the `ScalarImplementationHeader` seems not to be precise enough - I'm more than interested in listing to propositions of better names :-)